File: SYP.Web/Modules/Customer/Customers/Editor/CustomersEditorForm.cs
namespace SYP.Customer.Forms;

[FormScript("Customer.CustomersEditor")]
[BasedOnRow(typeof(CustomersRow), CheckNames = true)]
public class CustomersEditorForm
{
    public string Code { get; set; }
    public string Name { get; set; }
    public string Email { get; set; }
    public string Phone { get; set; }
    public string Address { get; set; }
    public bool IsActive { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Customer/CustomersEditorForm.ts
import { BooleanEditor, DateEditor, initFormType, IntegerEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface CustomersEditorForm {
    Code: StringEditor;
    Name: StringEditor;
    Email: StringEditor;
    Phone: StringEditor;
    Address: StringEditor;
    IsActive: BooleanEditor;
    InsertDate: DateEditor;
    InsertUserId: IntegerEditor;
    UpdateDate: DateEditor;
    UpdateUserId: IntegerEditor;
}

export class CustomersEditorForm extends PrefixedContext {
    static readonly formKey = 'Customer.CustomersEditor';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CustomersEditorForm.init) {
            CustomersEditorForm.init = true;

            var w0 = StringEditor;
            var w1 = BooleanEditor;
            var w2 = DateEditor;
            var w3 = IntegerEditor;

            initFormType(CustomersEditorForm, [
                'Code', w0,
                'Name', w0,
                'Email', w0,
                'Phone', w0,
                'Address', w0,
                'IsActive', w1,
                'InsertDate', w2,
                'InsertUserId', w3,
                'UpdateDate', w2,
                'UpdateUserId', w3
            ]);
        }
    }
}